test/test_handler.py

```diff
@@ -16,65 +17,84 @@
         self.assertEqual(response["statusCode"], "400")
 
     @patch("src.aws.lock.dynamodb_lock_client.acquire_lock")
     def test_swallows_dynamodb_lock_error(self, _mock_lock):
         _mock_lock.side_effect = DynamoDBLockError("error")
-        response = handle_github_webhook(
+        response = handler.handle_github_webhook(
             event_type="pull_request",
-            delivery_id="123",
             webhook_body=json.dumps(WEBHOOK_BODY_TEMPLATE),
-            should_retry=False,
         )
 
         self.assertEqual(response["statusCode"], "500")
         self.assertEqual(response["body"], "DynamoDBLockError: error - Unknown error")
 
     def test_throws_all_other_errors(self):
-        response = handle_github_webhook(
+        response = handler.handle_github_webhook(
             event_type="not_found",
-            delivery_id="123",
             webhook_body=json.dumps(WEBHOOK_BODY_TEMPLATE),
-            should_retry=False,
         )
 
         self.assertEqual(response["statusCode"], "501")
         self.assertEqual(response["body"], "No handler for event type not_found")
 
         with patch("src.github.webhook.handle_github_webhook") as mock:
             mock.side_effect = Exception("error")
-            response = handle_github_webhook(
+            response = handler.handle_github_webhook(
                 event_type="pull_request",
-                delivery_id="123",
                 webhook_body=json.dumps(WEBHOOK_BODY_TEMPLATE),
-                should_retry=False,
             )
 
         self.assertEqual(response["statusCode"], "500")
         self.assertEqual(response["body"], "error")
 
 
 class TestSQSRequeue(MockSQSTestCase):
+    @patch.object(hmac, "compare_digest", return_value=True)
     @patch("src.aws.lock.dynamodb_lock_client.acquire_lock")
-    def test_requeue_on_error(self, _mock_lock):
+    def test_requeue_on_error(self, _mock_lock, _mock_compare_digest):
         _mock_lock.side_effect = DynamoDBLockError("error")
         with patch("src.aws.sqs_client.SQS_URL", self.test_queue_url):
-            response = handle_github_webhook(
-                event_type="pull_request",
-                delivery_id="123",
-                webhook_body=json.dumps(WEBHOOK_BODY_TEMPLATE),
-                should_retry=True,
+            response = handler.handler(
+                event={
+                    "headers": {
+                        "X-GitHub-Event": "pull_request",
+                        "X-Hub-Signature": "sha1=1234",
+                    },
+                    "body": json.dumps(WEBHOOK_BODY_TEMPLATE),
+                },
+                context={},
             )
-
         self.assertEqual(response["statusCode"], "500")
 
-        messages = self.client.receive_message(QueueUrl=self.test_queue_url)["Messages"]
+        messages = self.client.receive_message(QueueUrl=self.test_queue_url).get(
+            "Messages", []
+        )
```

Comment:
hm might need to change this assert that queue_new_event was called instead

Reply:
This test case uses the `mock_sqs` class from `moto` and not the traditional patch/mock from unittest. I don't believe I can access the callstack of `queue_new_event` because it get's mocked at a layer deeper than the sqs_client